massa-final-state/src/final_state.rs

```diff
@@ -322,10 +356,19 @@ impl FinalState {
                 ))
             })?;
 
+            let mut batch = DBBatch::new();
+
             self.pos_state
-                .create_new_cycle_from_last(&latest_snapshot_cycle_info, first_slot, last_slot)
+                .create_new_cycle_from_last(
+                    &latest_snapshot_cycle_info,
+                    first_slot,
+                    last_slot,
+                    &mut batch,
+                )
                 .map_err(|err| FinalStateError::PosError(format!("{}", err)))?;
 
+            self.pos_state.db.write().write_batch(batch, Some(end_slot));
```

Comment:
The problem here is that for each new completed cycle, we feed the pos selector (that computes the draws). And this step needs to have a cycle written on disk.. We could implement some workaround, but I don't know what design is best.
For instance:
- Backup the DB before interpolating, and revert back to the initial DB if we crash during the interpolation.
- Tweak the pos functions to prevent panics if the cycles are not on disk (during this interpolation step, with a `during_interpolation: bool` passed around the selector for instance..

It's also kinda related to the optimization above (detailed in #4000).